### PR TITLE
Report: Evaluation (and some reflection)

### PR DESCRIPTION
**Pull request by casvdrest**
Length is just about 7 pages

### CODE REVIEW

File: report/report.tex [hunk 1]
```diff
@@ -26,24 +27,180 @@
...
...ariable. Local variables are declared in a \emph{local} statement with a name, type and default value, and undeclared in an \emph{unlocal} statement with a final value. By the time the unlocal statement is being executed, the local variable should have the specified final value. Note however, that the final value can also be a non-literal expression. The unlocal block is there to function as initial value for the local variable if the procedure that it is part of is being uncalled, i.e. reversed: in that case, the local variable needs what as declared as its final value as its initial value, and vice versa. By the time the unlocal block is reached, an assertion is executed to make sure that the procedure can safely be reverted without giving the local variable an incorrect initial value. An unlocal block is implemented as a \inlinecode{let-in} Haskell statement, because this automatically takes care of removing a local variable from scope after the unlocal line. To keep it simple, nesting multiple local blocks with identical variable names is impossible, and local variables can never have the name of a global variable. The let-in block again causes scoping issues, for example here:
```
Comment (jorisburgers) [inline report/report.tex]: This sentence is not wel formed: the local variable needs what as declared as its final value as its initial value, and vice versa.

File: report/report.tex [hunk 2]
```diff
@@ -26,24 +27,180 @@
...
+}
+procedure y() {
+    a += 1;
+    b += 1;
+}
+\end{minted}
+
+\noindent This procedure is compiled to something like the following:
+
+\begin{minted}{haskell}
```
Comment (jorisburgers) [inline report/report.tex]: I used the \code command from the poster in for my examples, maybe we should chose one or the other, but only one to remain consistent.

File: report/report.tex (hunk 1 above)
Comment (joristt) [reply report/report.tex]: Fixed

File: report/report.tex (hunk 2 above)
Comment (joristt) [reply report/report.tex]: There are already code examples from Ivo in the master branch that use minted. I suggest that we use minted for the smaller examles (including the ones in this PR) and use your fancy \code command for the larger examples that you wrote; it's a bit overkill to add line numbers and everyting to relatively small examples in my opinion.

**Comment by joristt** [comment]
@jorisburgers Looked at the sentences that refer to the AST, but they're not really sentences that require additional explanation of our AST structure.

I've fixed (or countered :p) all your suggestions now. #plzmerge

File: report/report.tex (hunk 2 above)
Comment (jorisburgers) [reply report/report.tex]: I'm fine with that solution

**Comment by jorisburgers** [comment]
@joristt My response was more in general that we maybe should have a small section to the AST, but I agree that your part specifically does not need any specific AST knowledge. I'll approve the request, but you have to fix the merge issues

**Comment by joristt** [comment]
Ah okay, fair enough. Don't think you can say a lot of interesting stuff about that though?

**Comment by jorisburgers** [comment]
@joristt No, not really, but I think there should be a few, maybe 3/4, lines on it

**Pull request merged**